### PR TITLE
How to create Django Docker images

### LINKED ISSUE #1101: How to containerize Django application using Docker
### Brief Summary: 
[Docker](https://docs.docker.com/get-started/overview/), an open-source project launched in 2013, has helped popularize technology and has helped drive the trend towards containerization and micro-services in software development that’s known as cloud-native development. In the article I will be explaining how to containerize the Todo application that was created in [this](https://www.section.io/engineering-education/django-crud-api/) article using Docker and push the Docker container to [Docker hub](https://hub.docker.com/).

### Key Takeaways: 
1. What is Docker.
2. What is Dockerfile and how to create it.
3. Creating Docker container from Dockerfile.
4. Pushing Docker container to the Docker hub.
### References: 
[Understanding Docker concepts](https://www.section.io/engineering-education/docker-concepts/)
[Getting started with Docker](https://www.section.io/engineering-education/getting-started-with-docker/)

### PR DESCRIPTION
**Pull request by paulodhiambo**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1101

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by geoffrey45** [comment]
@hectorkambow this article needs a plagiarism check.

**Comment by hectorkambow** [comment]
plagiarism check done

File: articles/django-docker/index.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+$ docker container ps
+CONTAINER ID   IMAGE                COMMAND                  CREATED       STATUS       PORTS                    NAMES
+d73306a9fb04   django_todo:latest   "python manage.py ru…"   3 hours ago   Up 3 hours   0.0.0.0:8000->8000/tcp   django_todo
+```
+On your browser visit `http://localhost:8000/api/v1/todo/` to confirm if the `django_todo` application is running in the container and you get a list of `todo` as a response.
+![Docker Hub image](/engineering-education/django-docker/docker-create.png)
+
+### Publishing Docker image to Docker Hub
+Docker Hub is a repository of container images that can be used to create other docker images.
+
```
Comment (geoffrey45) [inline articles/django-docker/index.md:166]: Docker images are used to create Docker containers. Please update your explanation.

File: articles/django-docker/index.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+```
+Move into the project directory by executing the command below.
+```bash
+$ cd django_todo
+```
+Execute the command below to install all the required dependencies for the project to run.
+
+```bash
+$ pip install -r requirements,txt
+```
```
Comment (geoffrey45) [inline articles/django-docker/index.md:50]: Replace the comma (,) in the word "requirements,txt" with a dot (.)

File: articles/django-docker/index.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    ├── manage.py  
+    ├── requirements.txt    # < Django dependencies list
+    └── django_todo    
+        ├── settings.py # Django settings file
+        ├── urls.py
+        └── wsgi.py
+```
+### Creating a Dockerfile
+Dockerfile is a text file that contains instructions on how the Docker image will be built. A Dockerfile contains the below directives.
+- **FROM** directive sets the base image from which the Docker container will be built.
```
Comment (geoffrey45) [inline articles/django-docker/index.md:76]: Add the article 'a' before the word 'Dockerfile' in line 75

File: articles/django-docker/index.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+Once you have created an account and logged in to Docker Hub, create a repository with the name `django_todo` and the description `Django Docker Image`.
+
+Now that we have created a repository in Docker Hub, to push the Docker image to the repository we have created we execute the command below.
+```bash
+$ docker login
+
+$ docker tag django_todo:latest <Docker Hub username>/django_todo:latest
+
+$ docker push <Docker Hub username>/django_todo:latest
+```
```
Comment (geoffrey45) [inline articles/django-docker/index.md:178]: Please indicate that everything under `< >` needs to be replaced with real values.

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@paulodhiambo Great article. I was looking at your article and came across a few minor issues. Please make the changes and ping me when you are done.

File: articles/django-docker/index.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+- `COPY ./requirements.txt /usr/src/app` copies the `requirements.txt` file into the work directory in the container.
+- `RUN pip install -r requirements.txt`
+ installs all the required modules for the `django_todo` application to run in the container.
+- `COPY . /usr/src/app` copies all the project source code to the working directory in the container.
+- `EXPOSE 8000` exposes port 8000 for access from other applications.
+- `CMD ["python", "manage.py", "runserver", "0.0.0.0:8000"]` sets the executable commands in the container.
+
+### Building Docker image
+To build the Docker image from the Dockerfile we created above, execute the command below.
+
```
Comment (geoffrey45) [inline articles/django-docker/index.md:122]: Add 'the' before the word 'Docker' in this heading.

**Comment by paulodhiambo** [comment]
@geoffrey45 I have made all the recommended changes.

**Review by geoffrey45 — APPROVED** [review]

**Comment by geoffrey45** [comment]
@hectorkambow technical review done. Ready for final review.

**Pull request merged**